File: logica-de-programacion_1/clase1/exercises/clase1.ts
(() => {
 function addition(numberOne: number, numberTwo: number): number {
  return numberOne + numberTwo;
 }
 function substract(numberOne: number, numberTwo: number): number {
  return numberOne - numberTwo;
 }
 function multiply(numberOne: number, numberTwo: number): number {
  return numberOne * numberTwo;
 }
 function division(numberOne: number, numberTwo: number): number {
  return numberOne / numberTwo;
 }
 function triangleArea(base: number, height: number): number {
  return (base * height) / 2
 }

 // de celsius a fahrenheit
 //°F = (°C x 9/5) + 32
 function converter( celsius:number ): number {
  let fahrenheit = ( celsius * 9/5) + 32
  return fahrenheit
 }

 //print result
function printResult(): void {
 console.log(((8 * 2) / 6) + 4)
}

//notes average
function gradeAverage(notes: Array<number>): number {
 let addition = 0
 for (let i = 0; i < notes.length; i++) {
   addition += notes[i]
 }
 return addition / notes.length
}

let notesList = [4.0, 2.7, 3.2]

 console.log(addition(8,9))
 console.log(substract(6,3))
 console.log(multiply(8,9))
 console.log(division(4,2))
 console.log(triangleArea(8, 9))
 console.log(`${converter(32)} °F`)
 printResult()
 console.log(gradeAverage(notesList))
})()

File: logica-de-programacion_1/clase1/exercises/clase1.py
def addittion(a, b):
 print(a + b)

def substract(a, b):
 print(a - b)

def multiply(a, b):
 print(a * b)

def division(a, b):
 if b == 0:
  print("Error: No se puede dividir por cero.")
 else:
  print(a / b)

def area(base, height):
 print((base * height) / 2)

print('1 -> Suma')
print('2 -> Resta')
print('3 -> Multiplicación')
print('4 -> División')
print('5 -> rectangle area')
user_option = int(input('Ingresa una opción: '))

if user_option == 1:
 number_one = int(input('Ingresa el primer número: '))
 number_two = int(input('Ingresa el segundo número: '))
 addittion(number_one, number_two)
elif user_option == 2:
 number_one = int(input('Ingresa el primer número: '))
 number_two = int(input('Ingresa el segundo número: '))
 substract(number_one, number_two)
elif user_option == 3:
 number_one = int(input('Ingresa el primer número: '))
 number_two = int(input('Ingresa el segundo número: '))
 multiply(number_one, number_two)
elif user_option == 4:
 number_one = int(input('Ingresa el primer número: '))
 number_two = int(input('Ingresa el segundo número: '))
 division(number_one, number_two)
elif user_option == 5:
 number_one = int(input('Ingresa el primer numero'))
 number_two = int(input('Ingresa el segunfo numero'))
 area(8, 9)
else:
 print('Opción incorrecta')